scipy/stats/_continuous_distns.py

```diff
@@ -10481,41 +10502,51 @@ def _shape_info(self):
         return [ib, ic]
 
     def _argcheck(self, b, c):
-        b = float(b)  # ensure float
         return (b != 0.) & (c > 1.)
 
     def _get_support(self, b, c):
         return self.a, c
 
     def _pdf(self, x, b, c):
+        b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
         return b * x**-(b+1) / (1 - 1/c**b)
 
     def _logpdf(self, x, b, c):
         return np.log(b) - np.log(-np.expm1(-b*np.log(c))) - (b+1)*np.log(x)
 
     def _cdf(self, x, b, c):
+        b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
         return (1 - x**-b) / (1 - 1/c**b)
```

Comment:
I suspect these were written as `1/c**b` instead of `c**-b` only to avoid the int to the negative int gotcha. Now that we are ensuring that `b` is floating point (because `x` is always floating point), we can write it the more natural way, `c**-b`.